### ISSUE TITLE
Typing jank on google.com search field with "uBlock-Origin-dev-filter – All Search Engines – Global"

### ISSUE THREAD

**Issue by Fanolian** [quenhus/uBlock-Origin-dev-filter#348]
This was originally filed in BMO as [Bug 1801597 - Typing jank on google.com search field with adblocking filter "uBlock-Origin-dev-filter – All Search Engines – Global"](https://bugzilla.mozilla.org/show_bug.cgi?id=1801597). uBO author gorhill suggested that I should report to you instead (extra info with some pointers [here](https://www.reddit.com/r/uBlockOrigin/comments/z3h2iw/where_should_i_report_this_external_filter_list/) and [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1801597#c7))

I copy the my BMO report here:

User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/109.0
Build ID: 20221120214001

### Steps to Reproduce

1. In a new profile, install [uBlock Origin](https://addons.mozilla.org/firefox/addon/ublock-origin/) (currently 1.45.2).
2. Subscribe [uBlock-Origin-dev-filter – All Search Engines – Global](https://github.com/quenhus/uBlock-Origin-dev-filter#import-into-ublock-origin)(AG). (Or use [this permalink url](https://raw.githubusercontent.com/quenhus/uBlock-Origin-dev-filter/7d76850954a2bbcfea9919584c101aff033cc5ed/dist/all_search_engines/global.txt))
3. Go to https://www.google.com/search?q=f. (So it skips the seemingly unaffected search field at google.com homepage.)
4. Press and hold a key to type rapidly into google.com search field. Adding some spaces inbetween seems to make the issue worse.

### Expected Results
Typing should be smooth and there should be little janks in performance profiles.

### Actual Results
There are some janks/slowness as shown in the attached video and profiles (attached in section Screenshot).
The situation, however, is much worse in my main profile. I can see the jank even if I type a search phrase normally.

### Additional Notes
This issue seems to be related to google.com's autocomplete feature. If I make an extremely long search phrase, autocomplete flyout will not be activiated and the jank is gone. The jank is back after clicking the X in the search field to clear the search phrase and type again.
However, Google will temporarily suspend autocomplete (`https://www.google.com/complete/search?q=*` requests return 403) for an unknown period of time (few hours max?) if I spam the autocomplete. I cannot reproduce the jank when autocomplete is suspended. Changing IP address or set up a new profile may lift the suspension.

### System configuration
OS version: Windows 10 64bit 22H2 19045.2251
GPU model: Nvidia GTX 760 driver 473.81 (Aug 02, 2022)
Number of cores: i7-4771 4 cores 8 threads
Amount of memory (RAM): 16GB
SSD drive
Tested uBO versions: 1.45.3b7, 1.45.2, 1.35.2 (earliest version available in AMO)
Tested Firefox versions: Nightly 2022-11-23, 2022-11-20, Firefox 107

### Performance profiles
Captured in Firefox 107, new profile, uBO 1.45.3b7 with AG: https://share.firefox.dev/3Eug06G
Disabling AG: https://share.firefox.dev/3hY0ZSX

### Video demonstrations of the issue
https://user-images.githubusercontent.com/771216/203749338-8fbdf4cb-8be9-4aee-91f3-2e16fc18b2f4.mp4
### Performance recording (profile)
Profile URL: These are the profiles as shown in the above video in a new Nightly profile with uBO 1.45.2:
1st [0:01 - 0:06], AG enabled: https://share.firefox.dev/3UYMj4H
2nd [0:18 - 0:21], reload google.com and try again: https://share.firefox.dev/3gnyUUS
3rd [0:37 - 0:42], AG disabled. No janks: https://share.firefox.dev/3V3ZuB4
4th [0:59 - 1:04], AG re-enabled. Janks again: https://share.firefox.dev/3V88ZyY
5th (not shown in the video), Accessibility is force-disabled to eliminate [bug 1800028](https://bugzilla.mozilla.org/show_bug.cgi?id=1800028): https://share.firefox.dev/3Gu6HX7

about:support of this profile: [uBO dev filter about support.txt](https://github.com/quenhus/uBlock-Origin-dev-filter/files/10086261/uBO.dev.filter.about.support.txt)

-------
https://user-images.githubusercontent.com/771216/203750719-032083d7-b893-40ec-a2cc-ac1c3bd78db3.mp4

The jank is much more noticeable in my main profile.
Granted there should are other factors causing the jank, but disabling AG vastly improves the jank to an acceptable level.

Profile captured during the video; AG enabled: https://share.firefox.dev/3ENTdUH
AG disabled (captured at a much later time and browser restarted, PIDs had changed): https://share.firefox.dev/3Aw8WWa

[[main profile] about support.txt](https://github.com/quenhus/uBlock-Origin-dev-filter/files/10086282/main.profile.about.support.txt)

--------
https://user-images.githubusercontent.com/771216/203756860-64705082-4e52-4ec7-bf84-c28e379c2eb4.mp4

I can still reproduce the jank with a [modern gaming laptop](https://www.gigabyte.com/Laptop/AORUS-15P--Intel-11th-Gen/sp#sp) using (essentially) the same profile as my main profile.

The laptop was plugged in with a high performance power profile. iGPU was used in the video but dGPU had the jank too.

Profile URLs:
1st, AG enabled: https://share.firefox.dev/3ErP8Es
2nd, AG disabled: https://share.firefox.dev/3U2qqQU
3rd, AG re-enabled: https://share.firefox.dev/3GDDvwZ

During the 3rd profile the jank was gone when autocompletes returned `400` (not `403`) if the search phrase was too long.

Thank you.

**Comment by quenhus** [comment]
Hi,

Thanks for your contribution.

To be honest, I don't really know if their is a uBlock solution to this issue. To resolve this, we should use a userscript (Tampermonkey) to remove bad links, rather than uBlock. But not everyone want to install another extension.

However, here is a PR based on gorhill4 response on Reddit : #349 

[PR feat/optimize_google_filter : uBlock-Origin-dev-filter – All Search Engines – Dev](https://raw.githubusercontent.com/quenhus/uBlock-Origin-dev-filter/feat/optimize_google_filter/dist/all_search_engines/all.txt)

Could you check if it is better?

**Comment by quenhus** [comment]
If anyone has propositions of better filters for other search engines, I am interested.

**Comment by Fanolian** [comment]
The issue is much improved with the new optimisation.


https://share.firefox.dev/3AJBD1N: This profile is captured from my main profile; google logged in (40 results per page). Compared with [this profile I quoted in first post](https://share.firefox.dev/3ENTdUH).
There are only barely noticeable janks if I type normally in Google.

https://share.firefox.dev/3OEbCH6: This is capture from main profile with google logged out (10 results per page).

Thanks for the fix.

p.s. I used https://raw.githubusercontent.com/quenhus/uBlock-Origin-dev-filter/dba8140344c7455b6e91cb95f7dd821668e33c5f/dist/all_search_engines/global.txt instead of All search engines - dev  for the above profiles for consistency with the original report.
